shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -671,25 +675,43 @@ public void registerOnBackInvokedCallback() {
    */
   @VisibleForTesting
   public void unregisterOnBackInvokedCallback() {
-    if (Build.VERSION.SDK_INT >= 33) {
+    if (Build.VERSION.SDK_INT >= 33 && onBackInvokedCallback != null) {
       getOnBackInvokedDispatcher().unregisterOnBackInvokedCallback(onBackInvokedCallback);
       hasRegisteredBackCallback = false;
     }
   }
 
-  private final OnBackInvokedCallback onBackInvokedCallback =
-      Build.VERSION.SDK_INT >= 33
-          ? new OnBackInvokedCallback() {
-            // TODO(garyq): Remove SuppressWarnings annotation. This was added to workaround
-            // a google3 bug where the linter is not properly running against API 33, causing
-            // a failure here. See b/243609613 and https://github.com/flutter/flutter/issues/111295
-            @SuppressWarnings("Override")
-            @Override
-            public void onBackInvoked() {
-              onBackPressed();
-            }
-          }
-          : null;
+  private final OnBackInvokedCallback onBackInvokedCallback = createOnBackInvokedCallback();
```

Comment:
It may be nice to write a test that verifies this callback responds to each method as we'd expect.

Reply:
done

Reply:
I don't believe that warning should cause that failure. We use compileSdk version 34 but the warning is just suggestion we upgrade our AGP version to accommodate that fact.

To get the test running against API 34, you need to annotate the test with `@Config(sdk = 34)`. Can you try making that change, then checking the type of `onBackInvokedCallback`?

Reply:
I've attempted to use the `@Config(sdk = 34)` annotation to test against API 34, but I encountered problems. In the meantime, I have created separate tests to cover different SDK versions and scenarios: `itHandlesOnBackAnimationCallbackAsExpectedForSdk34OrHigher`, `onBackInvokedCallbackCallsOnBackPressedForSdk33`, and `onBackInvokedCallbackIsNullForSdk32OrLower`.

Could you please take a look at the modifications I've made to the tests?

Reply:
The tests look good to me! What issues are you running into?

Looks like the 34 test may be failing because of the verifications for the callback in the test. I'm not familiar with `FlutterActivityWithReportFullyDrawn` so it may have to do with its implementation if you've double checked those are the method calls we need to verify there.